# Assignment 3 (Q4)

## Q4. TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [2]:
# Load the dataset
customer=pd.read_csv('Costomer+OrderForm.csv')
customer

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
5     Error Free  Error Free  Error Free  Error Free
6     Error Free   Defective  Error Free  Error Free
7     Error Free  Error Free  Error Free  Error Free
8     Error Free  Error Free  Error Free  Error Free
9     Error Free  Error Free  Error Free  Error Free
10     Defective  Error Free  Error Free  Error Free
11    Error Free  Error Free  Error Free  Error Free
12    Error Free  Error Free  Error Free   Defective
13    Error Free  Error Free  Error Free  Error Free
14    Error Free  Error Free  Error Free  Error Free
15    Error Free  Error Free   Defective  Error Free
16    Error Free  Error Free  Error Free  Error Free
17    Error Free  Error Free  Error Free  Error Free
18    Error Free  Error Free  Error Free  Error Free
19     Defective  Error Free  Error Free   Defective
20    Error Free  Error Free  Error Free  Error Free
21    Error Free  Error Free   Defective  Error Free
22    Error Free  Error Free  Error Free  Error Free
23     Defective  Error Free  Error Free  Error Free
24    Error Free  Error Free  Error Free  Error Free
25    Error Free  Error Free  Error Free  Error Free
26    Error Free  Error Free  Error Free  Error Free
27    Error Free  Error Free  Error Free  Error Free
28    Error Free  Error Free  Error Free  Error Free
29    Error Free  Error Free  Error Free  Error Free
..           ...         ...         ...         ...
270   Error Free  Error Free  Error Free  Error Free
271   Error Free  Error Free  Error Free  Error Free
272   Error Free  Error Free  Error Free  Error Free
273   Error Free  Error Free  Error Free  Error Free
274   Error Free   Defective   Defective  Error Free
275   Error Free  Error Free  Error Free  Error Free
276   Error Free  Error Free  Error Free  Error Free
277   Error Free  Error Free  Error Free  Error Free
278   Error Free  Error Free  Error Free  Error Free
279   Error Free  Error Free  Error Free  Error Free
280   Error Free  Error Free  Error Free  Error Free
281   Error Free  Error Free  Error Free  Error Free
282   Error Free  Error Free  Error Free  Error Free
283   Error Free   Defective  Error Free  Error Free
284   Error Free  Error Free  Error Free  Error Free
285   Error Free  Error Free  Error Free   Defective
286   Error Free  Error Free  Error Free  Error Free
287   Error Free  Error Free  Error Free  Error Free
288   Error Free  Error Free   Defective  Error Free
289    Defective  Error Free  Error Free  Error Free
290   Error Free  Error Free  Error Free  Error Free
291    Defective  Error Free  Error Free  Error Free
292   Error Free   Defective  Error Free  Error Free
293   Error Free  Error Free  Error Free  Error Free
294   Error Free   Defective  Error Free  Error Free
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [3]:
customer.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [4]:
customer.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [5]:
customer.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [6]:
customer.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [7]:
# Make dimensional array
obs=np.array([[271,267,269,280],[29,33,31,20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

## Assume Null Hypothesis as Ho: Independence of categorical variables (customer order forms defective % does not varies by centre) Thus, Alternative hypothesis as Ha Dependence of categorical variables (customer order forms defective % varies by centre) 

In [8]:
# Chi2 contengency independence test
chi2_contingency(obs) # o/p is (Chi2 stats value, p_value, df, expected obsvations)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [9]:
# Compare p_value with α = 0.05

## Inference: As (p_value = 0.2771) > (α = 0.05); We Failed to reject Null Hypthesis i.e. Independence of categorical variables Thus, customer order forms defective % does not varies by centre  